include/Value.h

```diff
@@ -814,8 +828,8 @@ struct value_type80 {
 	}
 
 public:
-	bool operator==(const value_type80 &rhs) const { return memcmp(value_, rhs.value_, 10) == 0; }
-	bool operator!=(const value_type80 &rhs) const { return memcmp(value_, rhs.value_, 10) != 0; }
+	bool operator==(const value_type80 &rhs) const { return std::memcmp(value_, rhs.value_, sizeof(T)) == 0; }
```

Comment:
This change looks scary. What if you later decide to make T equal in size to long double? You may then fail to notice that you're comparing padding bytes in 64-bit code.